packages/store/src/store.ts

```diff
@@ -27,7 +31,7 @@ export class Store {
    * Selects a slice of data from the store.
    */
   select<T>(selector: (state: any, ...states: any[]) => T): Observable<T>;
-  select(selector: string | any): Observable<any>;
+  select(selector: StateClass): Observable<any>;
   select(selector: any): Observable<any> {
```

Comment:
select<T = any, K = any>(selector: T): Observable<K>

Reply:
This comment seemed to conflict with your earlier suggestions. Could you please explain further?

Also I've added string back to the type union, though I still don't see what the use case for passing a string is, is there anything in the documentation that explains this?